scripts/Run-Self-Hosted-Runner-Test.ps1

```diff
@@ -0,0 +1,261 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+#
+# This script executes the provided test command, waits for <TestHangTimeout> (in seconds) and then triggers user and
+# kernel dumps if the test process is still running (This is typically the case when the test process is hung for some
+# reason).
+#
+# The user mode dump is created using 'procdump64.exe' and the kernel dump using 'notmyfault64.exe' utilities
+# respectively, both from the SysInternals Suite.  The script assumes the presence of these utilities under the current
+# working directory.
+#
+# (While 'procdump64.exe' also has the ability to generate a kernel dump, that dump is restricted to the kernel
+# portions of the user mode app's threads _only_ and does not provide any visibility into other kernel threads.  This
+# script therefore uses the 'notmyfault64.exe' tool to generate a 'true' kernel dump which captures the state of all
+# kernel threads and related data structures)
+#
+
+[CmdletBinding()]
+Param(
+    [Parameter(Mandatory = $true)] [string] $TestCommand,
+    [Parameter(Mandatory = $false)] [string] $TestArguments = "",
+    [Parameter(Mandatory = $false)] [int] $TestHangTimeout = 3600,
+    [Parameter(Mandatory = $false)] [string] $UserModeDumpFolder = "C:\Dumps",
+    [Parameter(Mandatory = $false)] [bool] $NeedKernelDump = $true
+)
+
+# Finds and returns the specified tool's location under the current directory. If not found, throws an exception.
+function GetToolLocationPath
+{
+    param(
+        [Parameter(Mandatory = $True)] [string] $ToolName
+    )
+
+    $ToolLocationPath = Get-ChildItem -Path "$Pwd" `
+        -Recurse -Filter "$ToolName" -ErrorAction SilentlyContinue | Select-Object -First 1
+    if ($ToolLocationPath -eq $null) {
+        $ErrorMessage = "*** ERROR *** $ToolName not found under $Pwd."
+        Write-Log $ErrorMessage
```

Comment:
Should you use your helper function ThrowWithErrorMessage ?

Reply:
fixed.